### DISCUSSION TITLE
QuickFilter in master-detail relationship

### DISCUSSION THREAD

**Discussion by ziaatrix** [serenity-is/Serenity#7215] [General]
Hi,

To activate quick filter in grid, you can use "QuickFilter". 

How about in master-detail relationship, in master grid I wanted to filter it based on a field from detail?

For example
Master is Order
Detail is Order Line Item

In Order Line Item there is Product
In Order Grid I wanted to add Product Filter

Any idea please?
 
![image](https://github.com/user-attachments/assets/0c4a9b12-c2c1-4043-accc-bed17cfbb2b1)

**Comment by ziaatrix** [comment]
@VictorTomaili  any idea please? Thanks.

**Reply by VictorTomaili** [reply]
"Contains Product in Details" filter does this in demo.

**Reply by ziaatrix** [reply]
@VictorTomaili 

Yes I found the "Contains Product in Details" in the demo "https://demo.serenity.is/Northwind/Order". Could you please provide to me the sample source code as there is no sample source code from the above link.

**Reply by VictorTomaili** [reply]
https://github.com/serenity-is/common-features

**Comment by ziaatrix** [comment]
@VictorTomaili 

Thank you so much. I followed the example however I am getting error.

I inherited the List request to add "ProductId"
![image](https://github.com/user-attachments/assets/d0d5bdb2-6b15-4fb9-bbe2-4dc1381c1388)

However this "ProductId" is not found in our ListHandler 
![image](https://github.com/user-attachments/assets/a3f965ca-9b2b-42c7-976f-170d3e3abe59)

![image](https://github.com/user-attachments/assets/ff1c8ad2-e961-4dab-899e-3c5bed4cae32)

**Reply by VictorTomaili** [reply]
change the MyRequest object to your request object

**Reply by ziaatrix** [reply]
@VictorTomaili 

Thank you so much, it is working now, this is what I missed.
